app/src/test/java/org/simple/clinic/registration/register/RegistrationLoadingScreenControllerTest.kt

```diff
@@ -21,61 +19,84 @@ import org.simple.clinic.user.registeruser.RegistrationResult.UnexpectedError
 import org.simple.clinic.widgets.ScreenCreated
 import org.simple.clinic.widgets.UiEvent
 
-@RunWith(JUnitParamsRunner::class)
 class RegistrationLoadingScreenControllerTest {
 
   private val userSession = mock<UserSession>()
   private val screen = mock<RegistrationLoadingScreen>()
-  lateinit var controller: RegistrationLoadingScreenController
-
   private val uiEvents = PublishSubject.create<UiEvent>()
 
+  private val controller = RegistrationLoadingScreenController(userSession)
+
   @Before
   fun setUp() {
-    controller = RegistrationLoadingScreenController(userSession)
-
     uiEvents
         .compose(controller)
         .subscribe { uiChange -> uiChange(screen) }
   }
 
-  @Parameters(method = "params for testing user registration")
   @Test
-  fun `when screen is created then user registration api should be called`(results: RegistrationResult) {
-    whenever(userSession.register()).doReturn(Single.just(results))
-    whenever(userSession.clearOngoingRegistrationEntry()).doReturn(Completable.complete())
+  fun `when retry button is clicked, then loader should show and register api should be called`() {
+    whenever(userSession.register()).doReturn(Single.just<RegistrationResult>(Success))
+    whenever(userSession.clearOngoingRegistrationEntry()).doReturn(Completable.never())
+
+    uiEvents.onNext(RegisterErrorRetryClicked)
+
+    verify(userSession).register()
+    verify(userSession).clearOngoingRegistrationEntry()
+    verify(screen).openHomeScreen()
+  }
+
+  @Test
+  fun `when screen is created, then the user registration api should be called`() {
+    // given
+    whenever(userSession.register()) doReturn Single.never()
 
+    // when
     uiEvents.onNext(ScreenCreated())
 
+    // then
     verify(userSession).register()
-    when (results) {
-      Success -> {
-        verify(userSession).clearOngoingRegistrationEntry()
-        verify(screen).openHomeScreen()
-      }
-      NetworkError -> verify(screen).showNetworkError()
-      UnexpectedError -> verify(screen).showUnexpectedError()
-    }
+    verifyNoMoreInteractions(userSession)
+    verifyZeroInteractions(screen)
+  }
+
+  @Test
+  fun `when the register user call succeeds, then clear registration entry and go to home screen`() {
```

Comment:
The test implies that the registration entry should be cleared, but we don't have a verification to assert this. Are we missing something?

Reply:
This is something explicit since the `openHomeScreen` method won't be invoked if the completable returned from `UserSession#clearOngoingRegistrationEntry` does not complete. But you're right about this not being explicit. I'll add a verification.